enterprise_access/apps/content_assignments/models.py

```diff
@@ -0,0 +1,101 @@
+"""
+Models for content_assignments
+"""
+from django.db import models
+from django_extensions.db.models import TimeStampedModel
+from simple_history.models import HistoricalRecords
+
+from enterprise_access.apps.subsidy_access_policy.models import SubsidyAccessPolicy
+
+from .constants import LearnerContentAssignmentStateChoices
+
+
+class AssignmentPolicy(TimeStampedModel):
+    """
+    Manage the creation and lifecycle of LearnerContentAssignments according to configurable rules.
+    """
+    subsidy_access_policy = models.ForeignKey(
+        SubsidyAccessPolicy,
+        related_name="assignment_policy",
+        on_delete=models.CASCADE,
+        db_index=True,
+        unique=True,
+    )
+    # TODO: Add fields representing different configurable rules below.
+
+    history = HistoricalRecords()
+
+
+class LearnerContentAssignment(TimeStampedModel):
+    """
+    Represent an assignment of a piece of content to a learner.
+
+    .. pii: The learner_email field stores PII.
+    .. pii_types: email_address
+    .. pii_retirement: local_api
+    """
+    assignment_policy = models.ForeignKey(
+        AssignmentPolicy,
+        related_name="assignments",
+        on_delete=models.CASCADE,
+        db_index=True,
+        help_text="AssignmentPolicy defining the lifecycle rules of this assignment.",
+    )
+    learner_email = models.EmailField(
+        null=False,
+        blank=False,
+        db_index=True,
+        help_text="Email of learner to assign content.",
+    )
+    lms_user_id = models.IntegerField(
+        null=True,
+        blank=True,
+        db_index=True,
+        help_text=(
+            "The id of the Open edX LMS user record with which this LearnerContentAssignment is associated. "
+            "This may be null at time of creation."
+        ),
+    )
+    content_key = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        db_index=True,
+        help_text=(
+            "The globally unique content identifier to assign to the learner.  Joinable with "
+            "ContentMetadata.content_key in enterprise-catalog."
+        ),
+    )
+    content_quantity = models.BigIntegerField(
+        null=False,
+        blank=False,
+        help_text="Cost of the content in USD Cents.",
+    )
+    state = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        choices=LearnerContentAssignmentStateChoices.CHOICES,
+        default=LearnerContentAssignmentStateChoices.ALLOCATED,
+        help_text=(
+            "The current state of the LearnerContentAssignment. One of: "
+            f"{[choice[0] for choice in LearnerContentAssignmentStateChoices.CHOICES]}"
```

Comment:
Also updated this from choice[1]  to choice[0] so that the actual database values are shown instead of the pretty names.